### ISSUE TITLE
PHP file with // prettier-ignore on method call with comments between params - Error: Comment "// Some comment" was not printed. Please report this error!

### ISSUE THREAD

**Issue by tonix-tuft** [prettier/plugin-php#1402]
**@prettier/plugin-php v0.14.0**
[Playground link](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCACsAOj4gWMCG3iyVAYXQBbH2wAFAAkA-BdBQxAOVwBjKABlGABPNWswbAgAWgAmJxBoAEdBeGgA12V-PTkFEGIRCBgytgq2KoMAXxYoCngYdBQ0eCxJUDZsGDaRTtrMMEEJvBAAHgB+YzAAHUwtmNkx7AACYIDPPexD4C2rzAPbg7mODxiDqUFMGIwb7F8oAFlYajoCgACgAlAdLjc7tCAPQw+7QOCILLwZSYfpQLbQ6EAEho8GwmQAfGx0TQkP8aEDgVjsXS4Qd4j8DjFfD4oCQDgBGWl06EbEAAQXubFcPm5AqYvL59PhTPZLLZHJgB2yB2IFGV3WlMv5IAAkphNSRuiKxarJTrdbcGfKoIq-MqDgBmK3WgWG406Chmtji10gN0HUEAbmuMpxbAOAF46QK7QcAG5yCICsNQ27tLZZzAgdpAA)

**Input:**
```php
<?php

class PHPClass {


    public function someMethod() {
        // prettier-ignore
        $this->anotherMethod(
            // Some comment 1
            "A param 1",
                // Some comment 2 indented
                "Indented param 2",
                // Some comment 3
                "Indented param 3"
        );

        $a =      "Some value";
    }
}

```

**Output:**
```php
Comment "// Some comment 1" was not printed. Please report this error!
```

**Expected output**:

```php
<?php

class PHPClass
{
    public function someMethod()
    {
        // prettier-ignore
        $this->anotherMethod(
            // Some comment 1
            "A param 1",
                // Some comment 2 indented
                "Indented param 2",
                // Some comment 3
                "Indented param 3"
        );

        $a = "Some value";
    }
}


```

**Comment by czosel** [comment]
Thanks for the bug report @tonix-tuft!

**Comment by wdoug** [comment]
I just ran into what is probably the same issue with comments in a prettier-ignored array ([demo](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCADsAOgBMIFjAht4slQGF0AW19sAAoAEoH4roKGIBxuAMZQAMowAJ5q1mDYEAC0TizQAI6C8NCBbsoBenIKIMQiEDClbOVslQYAvixQFPAw6Cho8FiSoGzYMK0iHTWYYILjeCAAPAD8xmAAOpibAPTbAAQAApCwCEhZ8MqYfVCbACQUbOx7ALx7ANqbe197AORsP99Ad9ngA+PYARiYn2+u1+HABQJeYIcUMwAF0ANybTYgNpAA)):

Input:
```php
<?php

// @prettier-ignore
$data = [
    'a'          => 1,
    // 'b'       => 2,
];
```
Result: `Comment "// 'b'       => 2," was not printed. Please report this error!`

**Comment by wdoug** [comment]
I just saw it in functions as well. It seems to be any comments inside a `prettier-ignored` segment.

**Comment by tonix-tuft** [comment]
Any chance this is going to be fixed? Thank you!

**Comment by czosel** [comment]
@tonix-tuft Currently I don't have very much time to work on this plugin and I'm also not too familiar with the way prettier-ignore is handled, so I can't give any guarantees. I totally agree that this is a serious issue and I'll try to take a look at it soon. If you or anyone else would like to contribute you're most welcome and I'd be happy to help you get started.

**Comment by czosel** [comment]
I just took a closer look. The issue occurs because we our parser includes the leading `// ` in a comment's value. This should be no problem because we can (and do) overwrite `hasPrettierIgnore` accordingly, but prettier core currently calls `isNodeIgnoreComment` as a escape hatch to avoid this issue - which currently can't be overwritten by plugins.

However, this problem should disappear automatically once https://github.com/prettier/prettier/pull/8777 has been released, which improves the handling of comments inside ignore blocks in prettier core. I was able to confirm this by installing the current `master` of prettier/prettier.

I'd suggest to leave this issue open until the fix in prettier core has been released (and this issue hopefully just disappears :upside_down_face:)

**Comment by tonix-tuft** [comment]
OK, thank you!

**Comment by czosel** [comment]
The issue seems to be resolved since prettier `2.1.0`, so I'll go ahead and close this.

**Comment by tonix-tuft** [comment]
Will take a look, thank you!
